mim_ocr/graph/builders.py

```diff
@@ -0,0 +1,180 @@
+from typing import Tuple
+from abc import ABC, abstractmethod
+from enum import Enum
+
+import numpy as np
+from igraph import Vertex, Graph, Edge
+
+from mim_ocr.data_model.box import Box
+from mim_ocr.graph.graph_utils import boxes_horizontal_distance, boxes_verdical_distance
+
+
+class Builder(ABC):
+    @classmethod
+    def get_box_for_vertex(cls, vertex: Vertex) -> Box:
+        return vertex["box"]
+
+
+class VertexBuilder(Builder):
+    @classmethod
+    @abstractmethod
+    def build(cls, graph: Graph, vertex: Vertex):
+        raise NotImplementedError
+
+
+class EdgeBuilder(Builder):
+    class HorizontalDirection(Enum):
+        LEFT = 0
+        SAME = 1
+        RIGHT = 2
+
+    class VerdicalDirection(Enum):
+        UP = 0
+        SAME = 1
+        DOWN = 2
+
+    @classmethod
+    def build(cls, graph: Graph, vertex1: Vertex):
+        for vertex2 in graph.vs:
+            if vertex1 == vertex2:
+                continue
+            edge_properties = cls._find_properties(vertex1, vertex2)
+            if len(edge_properties["directions"]) == 0:
+                continue
+            # builder_properties = {cls.__name__: edge_properties}
```

Comment:
Jeśli to pozostałość po nieaktualnym kodzie, to ją usuńmy.

Reply:
Jasne. usunąłem.